src/interpolate.jl

```diff
@@ -158,7 +158,10 @@ function unstructured_2d_to_3d(unstructured_data, coordinates, levels,
 
   center_level_0 = center_level_0[other_dimensions]
 
-  return unstructured_data, new_coordinates, new_levels, center_level_0
+  # Extract labels for plotting
+  plot_labels = ["x", "y", "z"][other_dimensions]
+
+  return unstructured_data, new_coordinates, new_levels, center_level_0, plot_labels
```

Comment:
TBH, I'd prefer to move this outside the 2D-to-3D conversion function. `unstructured_2d_to_3d` is right now a purely algorithmic function. Now you are adding functionality on top of that that is only relevant for the generated images. Thus I'd accept the (very minor) increase in code size by determining the label names in `convert.jl` directly.